src/services/leafletMapDefaults.js

```diff
@@ -101,69 +102,8 @@ angular.module('ui-leaflet').factory('leafletMapDefaults', function ($q, leaflet
         setDefaults: function (userDefaults, scopeId) {
             var newDefaults = _getDefaults();
 
-            if (isDefined(userDefaults)) {
```

Comment:
how about using angular.merge? https://docs.angularjs.org/api/ng/function/angular.merge
seems to be more safe with complex objects.